app/scripts/controllers/bridge-status/bridge-status-controller.ts

```diff
@@ -0,0 +1,316 @@
+  bridgeStatusState: BridgeStatusControllerState;
+}> = {
+  // We want to persist the bridge status state so that we can show the proper data for the Activity list
+  // basically match the behavior of TransactionController
+  bridgeStatusState: {
+    persist: true,
+    anonymous: false,
+  },
+};
+
+type SrcTxHash = string;
+export type FetchBridgeTxStatusArgs = {
+  statusRequest: StatusRequest;
+};
+export default class BridgeStatusController extends StaticIntervalPollingController<
+  typeof BRIDGE_STATUS_CONTROLLER_NAME,
+  { bridgeStatusState: BridgeStatusControllerState },
+  BridgeStatusControllerMessenger
+> {
+  #pollingTokensBySrcTxHash: Record<SrcTxHash, string> = {};
+
+  constructor({
+    messenger,
+    state,
+  }: {
+    messenger: BridgeStatusControllerMessenger;
+    state?: Partial<{
+      bridgeStatusState: BridgeStatusControllerState;
+    }>;
+  }) {
+    super({
+      name: BRIDGE_STATUS_CONTROLLER_NAME,
+      metadata,
+      messenger,
+      // Restore the persisted state
+      state: {
+        ...state,
+        bridgeStatusState: {
+          ...DEFAULT_BRIDGE_STATUS_CONTROLLER_STATE,
+          ...state?.bridgeStatusState,
+        },
+      },
+    });
+
+    // Register action handlers
+    this.messagingSystem.registerActionHandler(
+      `${BRIDGE_STATUS_CONTROLLER_NAME}:startPollingForBridgeTxStatus`,
+      this.startPollingForBridgeTxStatus.bind(this),
+    );
+    this.messagingSystem.registerActionHandler(
+      `${BRIDGE_STATUS_CONTROLLER_NAME}:wipeBridgeStatus`,
+      this.wipeBridgeStatus.bind(this),
+    );
+
+    // Set interval
+    this.setIntervalLength(REFRESH_INTERVAL_MS);
+
+    // If you close the extension, but keep the browser open, the polling continues
+    // If you close the browser, the polling stops
+    // Check for historyItems that do not have a status of complete and restart polling
+    this.restartPollingForIncompleteHistoryItems();
+  }
+
+  resetState = () => {
+    this.update((_state) => {
+      _state.bridgeStatusState = {
+        ...DEFAULT_BRIDGE_STATUS_CONTROLLER_STATE,
+      };
+    });
+  };
+
+  wipeBridgeStatus = ({
+    address,
+    ignoreNetwork,
+  }: {
+    address: string;
+    ignoreNetwork: boolean;
+  }) => {
+    // Wipe all networks for this address
+    if (ignoreNetwork) {
+      this.update((_state) => {
+        _state.bridgeStatusState = {
+          ...DEFAULT_BRIDGE_STATUS_CONTROLLER_STATE,
+        };
+      });
+    } else {
+      const { selectedNetworkClientId } = this.messagingSystem.call(
+        'NetworkController:getState',
+      );
+      const selectedNetworkClient = this.messagingSystem.call(
+        'NetworkController:getNetworkClientById',
+        selectedNetworkClientId,
+      );
+      const selectedChainId = selectedNetworkClient.configuration.chainId;
+
+      this.#wipeBridgeStatusByChainId(address, selectedChainId);
+    }
+  };
+
+  restartPollingForIncompleteHistoryItems = () => {
```

Comment:
Nit: rename this to #restartPollingForIncompleteHistoryItems bc it's private

Reply:
Done in [ce4ef37](https://github.com/MetaMask/metamask-extension/pull/27740/commits/ce4ef37425e0ad0c794b5480975eea3626200871)